File: 463-Island Perimeter/463.py
class Solution(object):

    def islandPerimeter(self, grid):
        """
        :type grid: List[List[int]]
        :rtype: int
        """
        ans = 0
        for row in grid:
            ex_row = [0] + row + [0]
            for i in range(len(row) + 1):
                ans += 1 if ex_row[i] is not ex_row[i + 1] else 0
        grid_t = map(list, (zip(*grid)))
        for row in grid_t:
            ex_row = [0] + row + [0]
            for i in range(len(row) + 1):
                ans += 1 if ex_row[i] is not ex_row[i + 1] else 0
        return ans

File: 463. Island Perimeter/463.cpp
class Solution {
public:
    int islandPerimeter(vector<vector<int>>& grid) {
        int ans = 0;
        vector<int> row;
        for (int i=0; i<grid.size(); ++i){
            row.clear();
            row.push_back(0);
            for (int j=0; j<grid[i].size(); ++j)
                row.push_back(grid[i][j]);
            row.push_back(0);
            for (int j=1; j<row.size(); ++j)
                ans += (row[j] != row[j-1]) ? 1 : 0;
        }
        for (int j=0; j<grid[0].size(); ++j){
            row.clear();
            row.push_back(0);
            for (int i=0; i<grid.size(); ++i)
                row.push_back(grid[i][j]);
            row.push_back(0);
            for (int i=1; i<row.size(); ++i)
                ans += (row[i] != row[i-1]) ? 1 : 0;
        }
        return ans;
    }
};